packages/main/src/SplitButton.js

```diff
@@ -0,0 +1,354 @@
+
+		/**
+		 * Indicates if there is SPACE key pressed
+		 * @private
+		 */
+		_spacePressed: {
+			type: Boolean,
+			noAttribute: true,
+		},
+
+		/**
+		 * Indicates if there is SHIFT or ESCAPE key pressed
+		 * @private
+		 */
+		_shiftOrEscapePressed: {
+			type: Boolean,
+			noAttribute: true,
+		},
+	},
+	events: /** @lends sap.ui.webcomponents.main.SplitButton.prototype */ {
+		/**
+		 * Fired when the user clicks on the text button.
+		 */
+		"click": {},
+
+		/**
+		 * Fired when the user clicks on the arrow button.
+		 */
+		"arrow-click": {},
+	 },
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * <code>ui5-split-button</code> enables users to trigger actions. It is constructed of two separate buttons
+ * <code>text button</code> and <code>arrow button</code> that can be activated by clicking or tapping, or by
+ * pressing certain keyboard keys - <code>Space</code> or <code>Enter</code> for text button,
+ * and <code>Arrow Down</code> or <code>Arrow Up</code> for arrow button.
+ *
+ * <h3>Usage</h3>
+ *
+ * <code>ui5-split-button</code> consists two separate buttons:
+ * <ul>
+ * <li>for the first one (text button) you can define some <code>text</code> or an <code>icon</code>, or both.
+ * Also, it is possible to define different icon for active state of this button - <code>activeIcon</code>.</li>
+ * <li>the second one (arrow button) consists only <code>slim-arrow-down</code> icon.</li>
+ * </ul>
+ * You can choose a <code>design</code> from a set of predefined types (the same as for ui5-button) that offer
+ * different styling to correspond to the triggered action. Both text and arrow buttons have the same design.
+ * <br><br>
+ * You can set the <code>ui5-split-button</code> as enabled or disabled. Both parts of an enabled
+ * <code>ui5-split-button</code> can be pressed by clicking or tapping it, or by certain keys, which changes
+ * the style to provide visual feedback to the user that it is pressed or hovered over with
+ * the mouse cursor. A disabled <code>ui5-split-button</code> appears inactive and any of the two buttons
+ * cannot be pressed.
+ * <br><br>
+ * There are separate events that are fired on activating of <code>ui5-split-button</code> parts:
+ * <ul>
+ * <li><code>click</code> for the first button (text button)</li>
+ * <li><code>arrow-click</code> for the second button (arrow button)</li>
+ * </ul>
+ *
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents/dist/SplitButton.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.SplitButton
+ * @extends UI5Element
+ * @tagname ui5-split-button
+ * @public
+ */
+class SplitButton extends UI5Element {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return SplitButtonCss;
+	}
+
+	static get template() {
+		return SplitButtonTemplate;
+	}
+
+	onAfterRendering() {
+		this.textButton.icon = this.buttonIcon;
```

Comment:
I somehow cancelled this comment in the previous review, so I'm adding it again.

Do not directly manipulate the DOM whenever possible, only use state. See my other comment about this. Basically, you are expected to create a private property `_textButtonState`, bind it to the button in the template and only change this property, which in turn will trigger invalidation and update the template.

Reply:
Done.